src/components/PostListVolunteer/index.jsx

```diff
@@ -0,0 +1,149 @@
+    if (e.target.value === "title") {
+      let res = await db
+        .collection("donatorPost")
+        .orderBy("title", "asc")
+        .get();
+      const data = res.docs.map((post) => {
+        const tmp = post.data();
+        const id = post.id;
+        return { id, ...tmp };
+      });
+      setPosts(data);
+    } else if (e.target.value === "date") {
+      let res = await db
+        .collection("donatorPost")
+        .orderBy("date", "desc")
+        .get();
+      const data = res.docs.map((post) => {
+        const tmp = post.data();
+        const id = post.id;
+        return { id, ...tmp };
+      });
+      setPosts(data);
+    }
+  };
+
+  const handleType = (e) => {
+    if (e.target.value === "all") {
+      setPosts(secPosts);
+    } else {
+      let fltr = secPosts.filter((el) => el.type === e.target.value);
+      setPosts(fltr);
+    }
+  };
+
+  const handleCity = async (e) => {
+    if (e.target.value === "all") {
+      setPosts(secPosts);
+    } else {
+      let fltr = secPosts.filter((el) => el.address[0] === e.target.value);
+      setPosts(fltr);
+    }
+  };
+
+  const handleSearch = async (e) => {
+    const search = e.target.value;
+
+    const result = secPosts.filter((post) =>
+      Object.values(post).find((el) => {
+        if (typeof el === "string") {
+          return el.toLowerCase().includes(search.toLowerCase());
+        }
+      })
+    );
+    setPosts(result);
+  };
+
+  return (
+    <Container fluid="lg">
+      <Row className="searchAndFilter">
+        <div>
+          <label>Type</label>
+          <select className="firstSelect" id="type" onChange={handleType}>
+            <option value="all">All</option>
+            <option value="goods">{t("postPage.goods")}</option>
+            <option value="money">{t("postPage.money")}</option>
+            <option value="service">{t("postPage.service")}</option>
+          </select>
+        </div>
+
+        <div>
+          <label>City</label>
+          <select className="firstSelect" id="city" onChange={handleCity}>
+            <option value="all">All</option>
+            {secPosts.map((e) => {
+              return <option value={e.address[0]}>{e.address[0]}</option>;
+            })}
+          </select>
+        </div>
+
+        <input
+          className="search"
+          type="search"
+          placeholder="  Search"
+          onChange={handleSearch}
+        />
+      </Row>
+      <Row className="displaySettings" md={6}>
+        <p>
+          {t("postPage.showing")} {posts.length} {t("postPage.results")}
+        </p>
+        <div>
+          <label id="labelSort">{t("postPage.sort")}</label>
+          <select className="sort" onChange={handleSort}>
+            <option value="title">{t("postPage.sortTitle")}</option>
+            <option value="date">{t("postPage.sortDate")}</option>
+          </select>
+        </div>
+      </Row>
+
+      <Row className="posts" fluid="md">
+        {posts.map((item) => (
```

Comment:
Instead of loading you can add some gif image that indicates loading, or you can keep it up to you, this is just a suggestion